src/lib/y2network/sysconfig/interfaces_reader.rb

```diff
@@ -111,27 +111,15 @@ def find_connections
 
       # Instantiates an interface given a hash containing hardware details
       #
-      # If there is not information about the type, it will rely on NetworkInterfaces#GetTypeFromSysfs.
-      # This responsability could be moved to the PhysicalInterface class.
-      #
-      # @todo Improve detection logic according to NetworkInterfaces#GetTypeFromIfcfgOrName.
-      #
       # @param data [Hash] hardware information
       # @option data [String] "dev_name" Device name ("eth0")
       # @option data [String] "name"     Device description
       # @option data [String] "type"     Device type ("eth", "wlan", etc.)
       def build_physical_interface(data)
         Y2Network::PhysicalInterface.new(data["dev_name"]).tap do |iface|
           iface.description = data["name"]
-          type = data["type"] || Yast::NetworkInterfaces.GetTypeFromSysfs(iface.name)
           iface.renaming_mechanism = renaming_mechanism_for(iface.name)
-          iface.type = case type
-          when nil then InterfaceType::ETHERNET
-          when ::String then InterfaceType.from_short_name(type)
-          when InterfaceType then type
-          else
-            raise "Unexpected value in interface type #{type.class.inspect}:#{type.inspect}"
-          end
+          iface.type = TypeDetector.type_of(iface.name)
```

Comment:
What about s390 devices. There does not have at all until enabled the group device.

Reply:
good question, but I have no clue.

Reply:
I suggest to postpone it till michal get back.

Reply:
Something like:

```
iface.type = InterfaceType.from_short_name(data["type"]) || TypeDetector.type_of(iface.name)
```

Originally it was ```type = data["type"] || Yast::NetworkInterfaces.GetTypeFromSysfs(iface.name)``` so, in most of the cases the type was obtained from hardware. The inference of types is mainly needed for connection_configs where the interface could not be obtained from the device information but from the configuration.